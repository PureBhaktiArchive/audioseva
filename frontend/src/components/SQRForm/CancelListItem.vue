<template>
  <v-list-group v-bind="$attrs" @click="$emit('click')">
    <template v-slot:activator>
      <v-list-item :style="styles">
        <v-list-item-content>
          <v-list-item-title :style="{ height: 'auto' }">
            {{ header }}
          </v-list-item-title>
        </v-list-item-content>
      </v-list-item>
    </template>
    <div :style="{ border: styles.border }" class="pa-1">
      <v-checkbox
        class="pa-2"
        @change="handleChange"
        :value="selected"
        :label="label"
      >
      </v-checkbox>
      <div v-if="selected">
        <v-textarea
          :placeholder="placeholder"
          label="Comment"
          outlined
          class="pa-2"
          :rules="rules"
          :value="value"
          @input="handleTextChange"
          filled
        >
        </v-textarea>
        <v-btn type="submit">Confirm</v-btn>
      </div>
    </div>
  </v-list-group>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component({
  name: 'CancelListItem',
})
export default class CancelListItem extends Vue {
  @Prop() header!: string;
  @Prop() label!: string;
  @Prop() styles!: { [key: string]: any };
  @Prop() placeholder!: string;
  @Prop() selected!: boolean;
  @Prop({ default: () => [] })
  rules!: any[];
  @Prop() value!: any;

  handleChange(selected: boolean) {
    this.$emit('update:selected', selected);
  }

  handleTextChange(text: string) {
    this.$emit('input', text);
  }
}
</script>

<style scoped></style>
