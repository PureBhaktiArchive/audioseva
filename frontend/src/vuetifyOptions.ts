export default {
  icons: {
    iconfont: 'fa',
    values: {
      plus: 'fas fa-plus',
      listening: 'fas fa-headphones',
      track: 'fas fa-cut',
      sound: 'fas fa-music',
      download: 'fas fa-file-download',
      delete: 'fas fa-trash',
      event: 'fas fa-calendar',
      undo: 'fas fa-undo',
      upload: 'fas fa-upload',
      check: 'fas fa-check',
    },
  },
} as any;
