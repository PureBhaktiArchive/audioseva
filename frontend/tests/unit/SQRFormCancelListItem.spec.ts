import CancelListItem from '@/components/SQRForm/CancelListItem.vue';
import vuetifyOptions from '@/vuetifyOptions';
import { createLocalVue, mount } from '@vue/test-utils';
import Vue from 'vue';
import Vuetify from 'vuetify';

const localVue = createLocalVue();

describe('SQRFormCancelListItem', () => {
  let vuetify: typeof Vuetify;
  let propsData: { [key: string]: any };

  const getWrapper = (props: any = {}) =>
    mount(CancelListItem, {
      localVue,
      vuetify,
      propsData,
      ...props,
    });

  const clickListItem = (wrapper: any) => {
    wrapper.find("[role='button']").trigger('click');
    return Vue.nextTick();
  };

  beforeEach(() => {
    vuetify = new Vuetify(vuetifyOptions);
    propsData = {
      header: 'CLICK HERE if you are unable to play or download the audio',
      label: "I'm unable to play or download the audio",
      placeholder:
        'Please describe the problem here, we will allot you new lectures shortly',
      styles: {
        backgroundColor: '#fcf8e3',
        color: '#8a6d3b',
        border: 'solid .2rem #faebcc',
        whiteSpace: 'none',
        width: '100%',
      },
      selected: false,
    };
  });

  it('should render', () => {
    const wrapper = getWrapper();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should emit click on list-group click', () => {
    const wrapper = getWrapper();
    clickListItem(wrapper);
    expect(wrapper.emitted().click).toBeTruthy();
  });

  it('should emit update:selected on checkbox selected', async () => {
    const wrapper = getWrapper();
    await clickListItem(wrapper);
    wrapper.find("input[type='checkbox']").setChecked();
    expect(wrapper.emitted()['update:selected']).toBeTruthy();
  });

  it('should show cancel inputs when checkbox is selected', async () => {
    const wrapper = getWrapper();
    await clickListItem(wrapper);
    expect(wrapper.find('textarea').exists()).toBe(false);
    expect(wrapper.find('button').exists()).toBe(false);
    wrapper.setProps({ selected: true });
    await Vue.nextTick();
    expect(wrapper.find('textarea').exists()).toBe(true);
    expect(wrapper.find('button').exists()).toBe(true);
  });

  it('should emit input event on textarea input', async () => {
    const wrapper = getWrapper();
    await clickListItem(wrapper);
    wrapper.setProps({ selected: true });
    await Vue.nextTick();
    wrapper.find('textarea').setValue('new text');
    // @ts-ignore
    expect(wrapper.emitted().input[0][0]).toEqual('new text');
  });
});
