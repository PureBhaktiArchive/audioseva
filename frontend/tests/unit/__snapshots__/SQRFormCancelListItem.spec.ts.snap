// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SQRFormCancelListItem should render 1`] = `
<div class="v-list-group">
  <div tabindex="0" aria-expanded="false" role="button" class="v-list-group__header v-list-item v-list-item--link theme--light">
    <div tabindex="-1" class="v-list-item theme--light" style="background-color: rgb(252, 248, 227); color: rgb(138, 109, 59); white-space: none; width: 100%; border: .2rem solid #faebcc;">
      <div class="v-list-item__content">
        <div class="v-list-item__title" style="height: auto;">
          CLICK HERE if you are unable to play or download the audio
        </div>
      </div>
    </div>
    <div class="v-list-item__icon v-list-group__header__append-icon"><i aria-hidden="true" class="v-icon notranslate fas fa-chevron-down theme--light"></i></div>
  </div>
  <!---->
</div>
`;
